programs/utilities/stat.sc

```diff
@@ -1,83 +1,391 @@
+        'print mined <block> <player>' -> ['printStatValue', 'mined'],
+        'print crafted <item> <player>' -> ['printStatValue', 'crafted'],
+        'print used <item> <player>' -> ['printStatValue', 'used'],
+        'print broken <item> <player>' -> ['printStatValue', 'broken'],
+        'print picked_up <item> <player>' -> ['printStatValue', 'picked_up'],
+        'print dropped <item> <player>' -> ['printStatValue', 'dropped'],
+        'print killed <entity> <player>' -> ['printStatValue', 'killed'],
+        'print killed_by <entity> <player>' -> ['printStatValue', 'killed_by'],
+        'print misc <misc_stat> <player>' -> ['printStatValue', 'custom'],
+        'print extra <extra_stat> <player>' -> ['printStatValue', 'extra'],
+        'print digs <dig> <player>' -> ['printStatValue', 'digs'],
+        'print combined <combined_stat> <player>' -> ['printStatValue', 'combined'],
+
+        'carousel start' -> 'startCarousel',
+        'carousel stop' -> 'stopCarousel',
+        'carousel interval' -> ['carouselInterval', null],
+        'carousel interval <seconds>' -> 'carouselInterval',
+        'carousel remove <index>' -> 'removeCarouselEntry',
+        'carousel add mined <block>' -> ['addCarouselEntry', 'mined'],
+        'carousel add crafted <item>' -> ['addCarouselEntry', 'crafted'],
+        'carousel add used <item>' -> ['addCarouselEntry', 'used'],
+        'carousel add broken <item>' -> ['addCarouselEntry', 'broken'],
+        'carousel add picked_up <item>' -> ['addCarouselEntry', 'picked_up'],
+        'carousel add dropped <item>' -> ['addCarouselEntry', 'dropped'],
+        'carousel add killed <entity>' -> ['addCarouselEntry', 'killed'],
+        'carousel add killed_by <entity>' -> ['addCarouselEntry', 'killed_by'],
+        'carousel add misc <misc_stat>' -> ['addCarouselEntry', 'custom'],
+        'carousel add extra <extra_stat>' -> ['addCarouselEntry', 'extra'],
+        'carousel add digs <dig>' -> ['addCarouselEntry', 'digs'],
+        'carousel add combined <combined_stat>' -> ['addCarouselEntry', 'combined'],
+        'carousel list' -> 'listCarouselEntries'
     },
     'arguments' -> {
-        'custom' -> {
+        'block' -> {
+            'type' -> 'term',
+            'suggestions' -> global_block_list,
+            'options' -> global_block_list,
+            'case_sensitive' -> false
+        },
+        'item' -> {
+            'type' -> 'term',
+            'suggestions' -> global_item_list,
+            'options' -> global_item_list,
+            'case_sensitive' -> false
+        },
+        'entity' -> {
+            'type' -> 'term',
+            'suggestions' -> global_entity_list,
+            'options' -> global_entity_list,
+            'case_sensitive' -> false
+        },
+        'misc_stat' -> {
+            'type' -> 'term',
+            'suggestions' -> keys(global_misc_stats),
+            'options' -> keys(global_misc_stats),

```

Comment:
For some reason, the app complains when I try to load saying I need a list here. I changed the line to `'options' -> map(keys(global_misc_stats), _),` and it started working. idk if `keys` starte returning an iterator isntead of a list or something, but in `extra_stat` the app doesn't complain, so idk what's up with this. Maybe to be safe just switch all keys in the options fields to lists using `map`?

Using carpet 1.4.66

Reply:
I know what the issue is but there is no way to fix it in Scarpet. Basically I'm using as the suggestions/options of an argument a list that is obtained by reading a file. The issue is that this file is downloaded from `resources`, and right now there is no way to run some code inside the config map, between requiring some resources and declaring the commands. So the first time the script is loaded the it won't find the file but it will download it, while the next times it will find the file without any issue.